python/GafferUI/EditScopeUI.py

```diff
@@ -248,3 +252,19 @@ def __navigationMenuDefinition( self ) :
 			)
 
 		return result
+
+	@staticmethod
+	def __userNodes( editScope ) :
+
+		nodes = []
+
+		for node in Gaffer.Node.RecursiveRange( editScope ) :
```

Comment:
We tend to use `MetadataAlgo.nodesWithMetadata()` for this sort of thing, since traversing a large graph in Python can be a bit slow. I guess EditScope contents are relatively light at the moment, but since folks can add their own nodes it might be worth going the faster route?

Reply:
Ahh good call, I hadn't seen that before, thanks. Switched implementation [here](https://github.com/GafferHQ/gaffer/pull/3880/commits/84412f2de2c9dc4af1a11ee03df95daa3d07787a_=)